Disable Series Data Entry when 'isCurrent' and 'isPrimary' can be misassigned

alesan99 opened the pull request:
Temporary fix for #5022 for Series Entry.

Disables Bulk Carry Forward Range if you have more than one record on the following dependent relationships:
- determinations
- determiners
- collectors
- funding agents (probably not possible here, Collecting Trips are not dependent)
- addresses (probably not possible here, Agents are not dependent)

Bulk Carry Forward Count is unchanged by this PR.

<img width="1701" height="697" alt="image" src="https://github.com/user-attachments/assets/bea48c99-60af-43d2-a1be-a397d9773c8e" />

### Checklist

- [x] Self-review the PR after opening it to make sure the changes look good and
      self-explanatory (or properly documented)
- [x] Add relevant issue to release milestone
- [ ] Add pr to documentation list
- [ ] Add automated tests

### Testing instructions

- Go to Data Entry for Collection Object
- Enable Bulk Carry Forward Range
- Verify bulk carry forward is disabled for the following cases:
- [ ] If you have more than 1 determination
- [ ] If you have more than 1 determiner in any determinations
- [ ] If you have more than  1 collector in a Collecting Event
- [ ] If your CO is part of a consolidated COG (Other COGTypes should work, and number of records doesn't matter for this).
- [ ] Make sure you can bulk carry forward normally if none of the above are true.
- [ ] Make sure Bulk Carry Forward works as expected on COs in non-consoldiated COGs.

pashiav reviewed (changes requested):
- [x] If you have more than 1 determination
- [x] If you have more than 1 determiner in any determinations
- [x] If you have more than  1 collector in a Collecting Event
- [ ] If you have more than 1 funding agent in a Collecting Trip
- [x] Make sure you can bulk carry forward normally if none of the above are true.
---
<img width="300" height="210" alt="Screenshot 2025-07-23 at 3 27 30 PM" src="https://github.com/user-attachments/assets/d71f333a-e909-4605-8789-f3b4d0214f93" />

**Bugs**

**1. Bulk carry is not disabled when there is more than 1 funding agent in a Collecting Trip.**
_Record set (not same as video, just similar on test panel):_ https://calvertmarinemuseumgen20250721-bulk-temp-fix.test.specifysystems.org/specify/view/collectionobject/26348/?recordsetid=166
https://github.com/user-attachments/assets/4a0e5365-324a-4c73-a0d0-6e869856aba1
_Collection:_ Invertebrates
_Username_: spadmin


https://github.com/user-attachments/assets/d39a0eb6-9a04-4fd5-9159-1693e14a75b4

alesan99 commented:
> **1. Bulk carry is not disabled when there is more than 1 funding agent in a Collecting Trip.** _Record set (not same as video, just similar on test panel):_ 

That's my mistake, Collecting Trips aren't dependent on Collecting Events, so they won't be cloned when you Carry Forward a Collection Object.
Updated the testing instructions!

---

Also pushed a fix for Consolidated COGs. The isPrimary checkbox would also get misassigned when the cojo would get cloned.
Added to the testing instructions 👍

<img width="768" height="233" alt="image" src="https://github.com/user-attachments/assets/24c221a2-3220-4393-9d62-8320659c4434" />

emenslin reviewed (approved):
- [x] If you have more than 1 determination
- [x] If you have more than 1 determiner in any determinations
- [x] If you have more than  1 collector in a Collecting Event
- [x] If your CO is part of a consolidated COG (Other COGTypes should work, and number of records doesn't matter for this).
- [x] Make sure you can bulk carry forward normally if none of the above are true.
- [x] Make sure Bulk Carry Forward works as expected on COs in non-consoldiated COGs.

---

Looks good, carry forward is blocked in all cases.

bhumikaguptaa reviewed (approved):
- [X] If you have more than 1 determination
- [X] If you have more than 1 determiner in any determinations
- [X] If you have more than  1 collector in a Collecting Event
- [X] If your CO is part of a consolidated COG (Other COGTypes should work, and number of records doesn't matter for this).
- [X] Make sure you can bulk carry forward normally if none of the above are true.
- [X] Make sure Bulk Carry Forward works as expected on COs in non-consoldiated COGs.

works as expected.
